config/100-api-serviceaccount.yaml

```diff
@@ -19,6 +19,20 @@ metadata:
   namespace: tekton-pipelines
 ---
 apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: tekton-results-api
+rules:
+  # Results API needs to be able to verify incoming auth tokens.
+  - apiGroups: ["authentication.k8s.io"]
+    resources: ["tokenreviews"]
```

Comment:
question: do we need these? My understanding is that tokenreviews and subjectaccessreviews are just request/response formats (like the interceptorrequest format) while the `rules` only apply to actual resources that the API server or the RBAC system knows about. (I don't actually know a ton about this so I could be totally off base here).

Reply:
Ok, so so looks like rules can apply to non CRD types too as in the file below. But I guess I still have the question: do we need these two roles to make the SAR /TokenReview calls?

Reply:
The request/response format would come into play if we were writing an admission control webhook to handle the auth. In this case, this is happening directly in the API Server.

TokenReviews only expose "`create`" (POST) method as the mechanism for issuing review requests. This RBAC Role controls who is allowed to make these calls (otherwise you get an error like `tokenreviews.authentication.k8s.io is forbidden: User "system:serviceaccount:tekton-pipelines:tekton-results-api" cannot create resource "tokenreviews" in API group "authentication.k8s.io" at the cluster scope`). SAR reviews are similar.

Reply:
ahh got it! Interesting!